llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -500,7 +500,12 @@ def tensor_hook(
 
         if dist.get_global_rank() == 0:
             log.debug('Saving Hugging Face checkpoint in global rank 0')
-
+            
+            if hasattr(original_model.config, 'use_cache'):
```

Comment:
This should not be set on the original model/config because you might continue training after this function is run. I'd be fine setting it for the new model/config that are getting saved out though.

Reply:
great point, thanks!